lib/src/convert_es6_proxy.html

```diff
@@ -0,0 +1,198 @@
+        },
+
+        'has' : function(target,prop) {
+          var len = ES6._dartArrayLength(instance);
+          var pos = +prop;
+          if (!isNaN(prop) && pos>=0 && pos < len ) {
+            return true;
+          }
+          return prop in target;
+        },
+
+        'getOwnPropertyDescriptor': function(target, prop) {
+          var len = ES6._dartArrayLength(instance);
+          var pos = +prop;
+          if (!isNaN(prop) && pos>=0 && pos < len ) {
+            return { configurable: true, enumerable: true, value: ES6._dartArrayGet(instance,pos) };
+          }
+          return Object.getOwnPropertyDescriptor(target,prop);
+        },
+
+        'set' : function(target,propertyName,value,proxy) {
+          if (propertyName.toString() == 'length') {
+            ES6._dartArraySetLength(instance,+value);
+          } else if (isNaN(propertyName.toString())) {
+            target[propertyName] = value;
+          } else {
+            ES6._dartArrayPut(instance,+propertyName,value);
+          }
+          return true;
+        },
+        'ownKeys' : function(target) {
+          var x = Object.getOwnPropertyNames(target).slice();
+          for (var i = 0 ;i<ES6._dartArrayLength(instance);i++) {
+            x.push(""+i);
+          }
+          return x;
+        }
+      });
+    };
+
+
+    /**
+     * Maps ES6JsProxy support
+     */
+
+    var mapCommonBehaviors = {
+      '__dartClass__' : function(instance) { return instance; },
+      '__isES6Proxy__' : function(instance) { return true; }
+    };
+
+    function getPropertyDescriptor(obj,prop) {
+      while(obj!=null) {
+        var pd = Object.getOwnPropertyDescriptor(obj,prop);
+        if (pd) {
+          return pd;
+        }
+        obj = obj.__proto__;
+      }
+      return undefined;
+    }
+
+    ES6.createES6JsProxyForMaps = function(instance) {
+      return new Proxy({},{
+        'has' : function(target,prop){
+          return ES6._dartKeys(instance).indexOf(prop) >= 0;
+        },
+        getOwnPropertyDescriptor: function(target, prop) {
+          if (ES6._dartKeys(instance).indexOf(prop) >= 0) {
+            return { configurable: true, enumerable: true, value: ES6._dartGet(instance,prop) };
+          }
+          return Object.getOwnPropertyDescriptor(target,prop);
+        },
+
+        'get' : function(target,propertyName,proxy) {
+
+          if (ES6._dartKeys(instance).indexOf(propertyName)>=0) {
+            return ES6._dartGet(instance,propertyName);
+          }
+
+          // target first! (because of '__proto__' , 'constructor' , and dart interop internals ... )
+          var _cache;
+          try {
+            var pd = getPropertyDescriptor(target,propertyName);
+            if (pd) {
+              if (pd.get) {
+               _cache = pd.get.call(proxy);
+               } else  {
+                 _cache = pd.value;
+               }
+             }
+          } catch (error) {}  // ignore errors
```

Comment:
Actually I don't remember, sorry. There's a chance it was only for cases where the getter will throw.
But I don't want to remove the `catch` because there could be some other reason that now I have forgotten. If you agree I'll just print the error to the console for now and leave the catch.